src/Microsoft.AspNet.IISPlatformHandler/IISPlatformHandlerMiddleware.cs

```diff
@@ -83,7 +83,8 @@ private static void UpdateRemoteIp(HttpContext httpContext)
             if (xForwardedForHeaderValue != null && xForwardedForHeaderValue.Length > 0)
             {
                 IPAddress ipFromHeader;
-                if (IPAddress.TryParse(xForwardedForHeaderValue[0], out ipFromHeader))
+                int? port;
```

Comment:
If you get a port set it to `httpContext.Connection.RemotePort`, otherwise clear remote port.